### PR TITLE
Add support for fullName to sdk-backend-spi

### PR DESCRIPTION
**Pull request by kandl**
- Tiger backend supports only fullName.
- Added otional fullName prop to avoid assigning fullName as a user firstName in tiger.

JIRA: RAIL-3068

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2848/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4474/ : FAILURE in 12m 09s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2852/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4478/ : SUCCESS in 12m 47s

File: libs/sdk-backend-spi/src/user/index.ts [hunk 1]
```diff
@@ -1,39 +1,64 @@
...
     email?: string;
 
     /**
-     * First name
+     * Full name.
+     *
+     * Note: This property has higher priority than firstName / lastName.
+     * As users can be stored differently across different backend implementations
+     * (some of them supports only fullName, and some of them only combination of firstName and lastName),
+     * we are putting here a contract, that every spi backend implementation have to fill this property,
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/user/index.ts:33]: Nitpick: this seems unnecessarily verbose, what about

```
Note: This property has higher priority than firstName / lastName.
Backend implementation MUST fill this property if user names are supported.
```

File: libs/sdk-backend-spi/src/workspace/users/index.ts [hunk 2]
```diff
@@ -25,12 +25,25 @@ export interface IWorkspaceUser {
      * Contact email of the user
      */
     email: string;
+
+    /**
+     * Full name.
+     *
+     * Note: This property has higher priority than firstName / lastName.
+     * As users can be stored differently across different backend implementations
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/users/index.ts:33]: See my previous comment

File: libs/sdk-backend-spi/src/user/index.ts [hunk 3]
```diff
@@ -1,39 +1,64 @@
...
 }
 
+/**
+ * Gets the user full name
+ *
+ * @param user - user to get full name of
+ * @returns the user full name
+ * @alpha
+ */
+export function getUserFullName(user: IUser): string | undefined {
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/user/index.ts:56]: Naming: we do not use the `get` prefix in other similar functions (like `insightTitle`). Should we use just `userFullName` here?

Comment (kandl) [reply libs/sdk-backend-spi/src/user/index.ts:56]: 🤦 oh right

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/2862/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4488/ : SUCCESS in 12m 19s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1131

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/839/ : SUCCESS in 53s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1103/ : SUCCESS in 32s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4489/ : SUCCESS in 12m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/211/ : SUCCESS in 16m 52s (non-voting)

**Pull request merged**